pkg/websocket/client.go

```diff
@@ -162,8 +163,10 @@ func (c *Client) Run(ctx context.Context) {
 			c.cfg.Log.WithFields(log.Fields{
 				"prefix": "websocket.Client.Run",
 			}).Debug("Resetting the connection")
+			c.isResetting = true
```

Comment:
I believe this will introduce a data race, since the field is read and written to from separate goroutines without other coordination (locks, channels). I'd expect it to therefore suffer similar problems to the deadlock.

Reply:
i mentioned that in my comment above. I'm going to think about what you said above, because I believe there are a few other race conditions currently that are less frequent.

Reply:
Boolean state was removed.